components/camera/client.go

```diff
@@ -228,6 +179,34 @@ func (c *client) Stream(
 	return stream, nil
 }
 
+func (c *client) Image(ctx context.Context, mimeType string, extra map[string]interface{}) ([]byte, ImageMetadata, error) {
+	ctx, span := trace.StartSpan(ctx, "camera::client::Image")
+	defer span.End()
+	expectedType, _ := utils.CheckLazyMIMEType(mimeType)
+
+	convertedExtra, err := goprotoutils.StructToStructPb(extra)
+	if err != nil {
+		return nil, ImageMetadata{}, err
+	}
+	resp, err := c.client.GetImage(ctx, &pb.GetImageRequest{
+		Name:     c.name,
+		MimeType: expectedType,
+		Extra:    convertedExtra,
+	})
+	if err != nil {
+		return nil, ImageMetadata{}, err
+	}
+
+	if expectedType != "" && resp.MimeType != expectedType {
+		c.logger.CDebugw(ctx, "got different MIME type than what was asked for", "sent", expectedType, "received", resp.MimeType)
+	} else {
+		resp.MimeType = mimeType
```

Comment:
Is this redundant? Shouldnt we always fill with the response mime type? What should we do if response mimeType is empty?

Reply:
We use the word "hint" a lot when dealing with mime types. I think that's probably why we don't care about if its empty. As to why we don't always fill with the response mime type... I'm not sure

Reply:
I see that this is maintaining existing behavior, but if I'm reading this right, if the user asked for a `png` and the camera responded with a `jpeg` this code will return an `ImageMetadata{MimeType: "png"}` despite the bytes containing a jpeg, is that right?

No need to change right now, I'm just trying to understand.